src/LazyLoad.jsx

```diff
@@ -35,7 +35,11 @@ export default class LazyLoad extends Component {
 
     if (top === 0 || (top <= (scrollTop + window.innerHeight + threshold)
                       && (top + height) > (scrollTop - threshold))) {
-      this.setState({ visible: true });
+      this.setState({ visible: true }, () => {
+        if (onContentVisible) {
```

Comment:
This is not declarative way. There is difference between manually triggering event callback and listening changes, comparing current and previous states and trigger callback when they have been changed. In second case we have 2 decoupled parts. The part which is responsible for update visible state shouldn't know about triggering some events outside, but if you want you can dedicate this logic in an another place (didUpdate hook)